operator/pkg/ciliumendpointslice/reconciler.go

```diff
@@ -196,13 +216,149 @@ func (r *reconciler) reconcileCESDelete(ces *cilium_v2a1.CiliumEndpointSlice) (e
+			logfields.K8sNamespace, pod.Namespace,
+			logfields.Error, err,
+		)
+		return nil
+	}
+
+	networking, err := GetPodEndpointNetworking(pod, r.logger)
+	if err != nil {
+		r.logger.Debug("Could not get pod's endpoint networking",
+			logfields.K8sPodName, pod.GetName(),
+			logfields.K8sNamespace, pod.Namespace,
+			logfields.Error, err,
+		)
+		return nil
+	}
+
+	ciliumNode, err := r.ciliumNodeStore.ByIndex(op_k8s.CiliumNodeIPIndex, pod.GetHostIP())
+	if err != nil || ciliumNode == nil || len(ciliumNode) == 0 {
+		r.logger.Debug("Pod's host node has no associated CiliumNode",
+			logfields.K8sPodName, pod.GetName(),
+			logfields.K8sNamespace, pod.Namespace,
+			logfields.HostIP, pod.GetHostIP(),
+		)
+		return nil
+	}
+
+	namedPorts := make(models.NamedPorts, 0)
+	for _, container := range pod.Spec.Containers {
+		for _, port := range container.Ports {
+			if port.Name == "" {
+				continue
+			}
+
+			proto, err := getProtocolString(port.Protocol)
+			if err != nil {
+				continue
+			}
+
+			p := &models.Port{
+				Name:     port.Name,
+				Protocol: proto,
+				Port:     uint16(port.ContainerPort),
+			}
+			namedPorts = append(namedPorts, p)
+		}
+	}
+
+	return &cilium_v2a1.CoreCiliumEndpoint{
+		Name:       pod.GetName(),
+		IdentityID: identityId,
+		Networking: networking,
+		Encryption: ciliumNode[0].Spec.Encryption,
+		NamedPorts: namedPorts,
+	}
+}
+
+// Convert a slim_corev1.Protocol to models.PortProtocol format.
+func getProtocolString(p slim_corev1.Protocol) (string, error) {
+	switch p {
+	case slim_corev1.ProtocolTCP:
+		return models.PortProtocolTCP, nil
+	case slim_corev1.ProtocolUDP:
+		return models.PortProtocolUDP, nil
+	case slim_corev1.ProtocolSCTP:
+		return models.PortProtocolSCTP, nil
+	default:
+		return "", fmt.Errorf("unknown protocol: %s", p)
+	}
+}
+
+// Constructs an EndpointNetworking object for a given pod.
+func GetPodEndpointNetworking(pod *slim_corev1.Pod, logger *slog.Logger) (*cilium_v2.EndpointNetworking, error) {
+	addressPair := &cilium_v2.AddressPair{}
+
+	if len(pod.Status.PodIPs) == 0 {
+		return nil, fmt.Errorf("no IPs allocated to pod yet: %s", pod.GetName())
+	}
+
+	for _, podIP := range pod.Status.PodIPs {
+		ip, err := netip.ParseAddr(podIP.IP)
+		if err != nil {
+			logger.Debug("podIP error",
+				logfields.K8sPodName, pod.GetName(),
+				logfields.K8sNamespace, pod.Namespace,
+				logfields.IPAddr, podIP.IP,
+				logfields.Error, err,
+			)
+			return nil, err
+		}
+
+		if ip.Is4() {
+			addressPair.IPV4 = ip.String()
+		} else if ip.Is6() {
+			addressPair.IPV6 = ip.String()
+		}
+	}
+
+	if pod.GetHostIP() == "" {
+		return nil, fmt.Errorf("no hostIP for pod yet: %s", pod.GetName())
+	}
+
+	networking := &cilium_v2.EndpointNetworking{
+		Addressing: cilium_v2.AddressPairList{
+			addressPair,
+		},
+		NodeIP: pod.GetHostIP(),
+	}
+	return networking, nil
+}
+
+// Get the identity ID for a given pod.
+func (r *reconciler) getPodIdentityID(pod *slim_corev1.Pod) (int64, error) {
```

Comment:
Should CID updates enqueue an updates for all CES that reference that identity? Otherwise the risk is that a CES will continue referencing a stale CID that no longer exists (e.g., got manually deleted and then re-created).

Reply:
+1

This needs to reconcile for CID objects. As part of another commit, we need to maintain cid->ces mapping.